docker/configure_workers_and_start.py

```diff
@@ -469,32 +607,199 @@ def generate_worker_files(
+        for worker in workers_to_combo_list:
+            # Verify this is a real defined worker type. If it's not, stop everything so
+            # it can be fixed.
+            copy_of_template_config = WORKERS_CONFIG.get(worker)
+            if copy_of_template_config:
+                # So it's not a reference pointer
+                copy_of_template_config = copy_of_template_config.copy()
+            else:
+                error(
+                    worker
+                    + " is an unknown worker type! Was found in "
+                    + worker_type
+                    + ". Please fix!"
+                )
+
+            # Merge worker type template configuration data.
+            worker_config = merge_worker_template_configs(
+                worker_config, copy_of_template_config
+            )
+
+            # Check if there is to many of a worker there can only be one of.
+            # Will error and stop if it is a problem, e.g. 'background_worker'.
+            if worker in worker_type_shard_counter:
+                if not is_sharding_allowed_for_worker_type(worker):
+                    error(
+                        "There can be only a single worker with "
+                        + worker
+                        + " type. Please recount and remove."
+                    )
+            # Not in shard counter, add it. Don't need return value
+            increment_counter(worker_type_shard_counter, worker)
+
+        # Name workers by their type or requested name concatenated with an
+        # incrementing number. e.g. federation_reader1 or event_creator+event_persister1
+        if requested_worker_name:
+            worker_base_name = requested_worker_name
+            # It'll be useful to have this in the log in case it's a complex of many
+            # workers merged together. Note for Complement: it would only be seen in the
+            # logs for blueprint construction(which are not collected).
+            log(
+                "Worker name request found: "
+                + requested_worker_name
+                + ", for: "
+                + str(workers_to_combo_list)
+            )
+
         else:
-            error(worker_type + " is an unknown worker type! Please fix!")
+            # The worker name will be the worker_type, however if spaces exist
+            # between concatenated worker_types and the "+" because of readability,
+            # it will error on startup. Recombine worker_types without spaces and log.
+            if len(workers_to_combo_list) > 1:
+                worker_base_name = "+".join(workers_to_combo_list)
+                if worker_base_name != worker_type:
+                    log(
+                        "Default worker name would have contained spaces, which is not "
+                        "allowed(" + worker_type + "). Reformed name to not contain "
+                        "spaces: " + worker_base_name
+                    )
```

Comment:
Would it be better to just strip out the spaces in the worker_type string(if they are even there), or replace them with underscores? Spaces are a no-no there, as it will lead to a `FileNotFound` error when loading the worker.yaml files. This is barely an edge case, for what it's worth, and only allows flexibility and readability for humans when writing out the environmental variable.

```suggestion
            # The worker name will be the worker_type, however if spaces exist
            # between concatenated worker_types and the "+" because of readability,
            # it will error on startup with a 'FileNotFound' error. Replace spaces with 
            # underscores in the string worker_types and log.
            if len(workers_to_combo_list) > 1:
                if " " in worker_type:
                    worker_base_name = worker_type.replace(" ", "_")
                    log(
                        f"Default worker name would have contained spaces, which is not "
                        "allowed({worker_type}). Reformed name to not contain spaces: "
                        "{worker_base_name}"
                    )
```